src/coordination/coordinator_instance.cpp

```diff
@@ -487,108 +585,83 @@ auto CoordinatorInstance::ForceResetCluster_() -> ForceResetClusterStateStatus {
 
   if (!new_main.PromoteToMain(new_uuid, std::move(repl_clients_info), &CoordinatorInstance::MainSuccessCallback,
                               &CoordinatorInstance::MainFailCallback)) {
-    spdlog::warn("Force reset failed since promoting replica to main failed.");
-    return ForceResetClusterStateStatus::RPC_FAILED;
+    spdlog::warn("Exiting VerifyOrCorrectClusterState. Force reset failed since promoting replica to main failed.");
+    return VerifyOrCorrectClusterStateStatus::FAIL;
   }
   spdlog::trace("Promoted instance {} to main", new_main.InstanceName());
```

Comment:
Again, could we get hurt if main instance died in between?

Reply:
I hope you understand what I mean, the instance liveness could change since the time we query it until the time we use this info

Reply:
resolved